src/components/App.js

```diff
@@ -1,39 +1,55 @@
 import React from 'react';
 import { hot } from 'react-hot-loader/root';
-import { checkURIforError, getTokenFromURI, getAuthURL } from '../config/authConfig';
-import macaroon from '../assets/img/Macaroonicon.png';
+import { checkURIforError, getTokenFromURI, getTokenFromLocal } from '../config/authConfig';
+import { Dashboard } from './Dashboard';
+import { Sidebar } from './Sidebar';
+import { Welcome } from './Welcome';
+import { LoginLink } from './LoginLink';
+import UserPlaylists from './UserPlaylists';
+import CurrentPlaylist from './CurrentPlaylist';
 
 
 class App extends React.Component {
   state = {
     token: '',
-    user: '',
-    error: false,
-    errorMessage: '',
+    user: {
+      name: '',
+      id: '',
+      error: false,

```

Comment:
Having 2 properties for errors, in this case, isn't totally necessary. if `errorMessage` has a truthy value, then you know you have an error, which makes `error` redundant. I'd kill `errorMessage` and  just keep `error`, but store the error message instead of the boolean.